Servers/Administration/OJS.Servers.Administration/Controllers/ContestsController.cs

```diff
@@ -111,19 +119,28 @@ public IActionResult Problems([FromQuery] IDictionary<string, string> complexId)
             EntityAction action,
             IDictionary<string, string> entityDict)
         {
-            if (entity.CategoryId.HasValue)
+            if (action == EntityAction.Create)
             {
-                await this.contestCategoriesValidationHelper
-                    .ValidatePermissionsOfCurrentUser(entity.CategoryId.Value)
-                    .VerifyResult();
+                return;
             }
 
-            if (action != EntityAction.Create)
+            var userHasContestRightsValidationResult = await this.contestsValidationHelper
+                .ValidatePermissionsOfCurrentUser(entity.Id);
+
+            var userHasContestRights = userHasContestRightsValidationResult.IsValid;
+
+            var userHasCategoryRightsValidationResult = await this.contestCategoriesValidationHelper
+                .ValidatePermissionsOfCurrentUser(entity.CategoryId!.Value);
+
+            var userHasCategoryRights = userHasCategoryRightsValidationResult.IsValid;
+
+            if (userHasContestRights || userHasCategoryRights)
             {
-                await this.contestsValidationHelper
-                    .ValidatePermissionsOfCurrentUser(entity.Id)
-                    .VerifyResult();
+                return;
             }
+
+            userHasContestRightsValidationResult.VerifyResult();
+            userHasCategoryRightsValidationResult.VerifyResult();
```

Comment:
This seems like a useful function for validating Problems/Tests/Submissions/.. by just providing their `contest.Id`. I didn't see such logic, shouldn't we validate in Edit/Create views/actions as well?

Reply:
I've updated the code so it reuses `contestsValidationHelper.ValidatePermissionsOfCurrentUser`